wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -535,9 +551,28 @@ class PlatformRepo private constructor(val walletApplication: WalletApplication)
     //
     suspend fun addWalletAuthenticationKeysAsync(seed: DeterministicSeed, keyParameter: KeyParameter?) {
         withContext(Dispatchers.IO) {
-            val wallet = walletApplication.wallet
+            val wallet = walletApplication.wallet!!
             // this will initialize any missing key chains
-            wallet!!.initializeAuthenticationKeyChains(seed, keyParameter)
+            val authenticationGroupExtension = AuthenticationGroupExtension(wallet)
+
+            if (keyParameter != null) {
+                authenticationGroupExtension.addEncryptedKeyChains(
+                    wallet.params,
+                    seed,
+                    keyParameter,
+                    keyChainTypes
+                )
+            } else {
+                // TODO: do we need this branch? addWalletAuthenticationKeysAsync is only called with non-null keyParameter
+                authenticationGroupExtension.addKeyChains(
+                    wallet.params,
+                    seed,
+                    keyChainTypes
+                )
+            }
```

Comment:
I suppose we don't need this `else`.  We have other places in DashPay that handle the case of a `null` `keyParameter`, but the wallet is always encrypted so this `keyParameter` will always be non-null.